src/bun.js/bindings/KeyObject.cpp

```diff
@@ -0,0 +1,2352 @@
+                        if (impl->type() != CryptoKeyType::Private) {
+                            auto scope = DECLARE_THROW_SCOPE(vm);
+                            throwException(globalObject, scope, createTypeError(globalObject, "Invalid private key"_s));
+                            return JSValue::encode(JSC::jsUndefined());
+                        }
+                        return JSC::JSValue::encode(JSCryptoKey::create(structure, zigGlobalObject, WTFMove(impl)));
+                    } else if (jwk.kty == "RSA"_s) {
+                        auto result = CryptoKeyRSA::importJwk(CryptoAlgorithmIdentifier::RSA_OAEP, std::nullopt, WTFMove(jwk), true, jwk.usages);
+                        if (result == nullptr) {
+                            auto scope = DECLARE_THROW_SCOPE(vm);
+                            throwException(globalObject, scope, createTypeError(globalObject, "Invalid RSA private key"_s));
+                            return JSValue::encode(JSC::jsUndefined());
+                        }
+                        auto impl = result.releaseNonNull();
+                        if (impl->type() != CryptoKeyType::Private) {
+                            auto scope = DECLARE_THROW_SCOPE(vm);
+                            throwException(globalObject, scope, createTypeError(globalObject, "Invalid private key"_s));
+                            return JSValue::encode(JSC::jsUndefined());
+                        }
+                        return JSC::JSValue::encode(JSCryptoKey::create(structure, zigGlobalObject, WTFMove(impl)));
+                    } else {
+                        auto scope = DECLARE_THROW_SCOPE(vm);
+                        throwException(globalObject, scope, createTypeError(globalObject, "Unsupported private key"_s));
+                        return JSValue::encode(JSC::jsUndefined());
+                    }
+                }
+                auto scope = DECLARE_THROW_SCOPE(vm);
+                JSC::throwTypeError(globalObject, scope, "The \"key\" property must be of type object"_s);
+                return JSValue::encode(JSC::jsUndefined());
+            }
+            }
+
+            auto format = formatJSValue.toWTFString(globalObject);
+            if (format == "jwk"_s) {
+                auto scope = DECLARE_THROW_SCOPE(vm);
+                JSC::throwTypeError(globalObject, scope, "The \"key\" property must be of type object"_s);
+                return JSC::JSValue::encode(JSC::JSValue {});
+            }
+
+            if (UNLIKELY(!data) || UNLIKELY(!byteLength)) {
+                auto scope = DECLARE_THROW_SCOPE(vm);
+                throwException(globalObject, scope, createTypeError(globalObject, "ERR_INVALID_ARG_TYPE: expected key to be Buffer or array-like object"_s));
+                return JSValue::encode(JSC::jsUndefined());
+            }
+
+            JSValue passphraseJSValue = options->getDirect(vm, PropertyName(Identifier::fromString(vm, "passphrase"_s)));
```

Comment:
This will crash if `passphrase` is not defined or if `passphrase` is on a prototype

Use `getIfPropertyExists` instead.